api/src/main/java/io/cloudevents/CloudEventAttributes.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2018-Present The CloudEvents Authors
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.cloudevents;
+
+import io.cloudevents.lang.Nullable;
+
+import javax.annotation.ParametersAreNonnullByDefault;
+import java.net.URI;
+import java.time.ZonedDateTime;
+import java.util.Set;
+
+/**
+ * The marker interface for CloudEvents attributes
```

Comment:
We should rephrase it as it is clearly not a marker interface. Consider the following:
```
Strategy which defines CloudEvent attributes as per specification - https://github.com/cloudevents/spec/blob/v1.0/spec.md#context-attributes
```

Reply:
I don't wanna touch any javadocs in this pr, i'm planning to do it in future as soon as we have other things aligned

Reply:
Most of the javadocs here are wrong or missing, so let's just focus on the interface now

Reply:
I understand your concern about some lingering javadocs, but this javadoc is clearly a new javadoc created as part of this PR so let's fix it here is well. 
No need to deliberately introduce something new that is already acknowledged by both of us as "wrong" with expectations that it is going to get fixed some time in the future.

Reply:
Now it should be fine

Reply:
let me know if it needs more refinement

Reply:
all good